feat(genapi): init docs (#3636)
scaleway/docs-content: ai-data/generative-apis/api-cli/using-generative-apis.mdx

tags: generative-apis ai-data embeddings-api
dates:
  validation: 2024-08-28
  posted: 2024-08-28
---

## Access

- Access to this service is restricted while in beta. You can request access to the product via [a form here](https://www.scaleway.com/en/betas/#generative-api).
- A valid [API key](/identity-and-access-management/iam/how-to/create-api-keys/) is needed.

## Authentication

All requests to the Scaleway Generative APIs must include an `Authorization` HTTP header with your API key prefixed by `Bearer`.

We recommend exporting your secret key as an environment variable, which you can then pass directly in your curl request as follows. Remember to replace the example value with your own API secret key.

```
export SCW_SECRET_KEY=720438f9-fcb9-4ebb-80a7-808ebf15314b
```

Curl request:

```
curl -X GET \
    -H "Authorization: Bearer ${SCW_SECRET_KEY}" \
    -H "Content-Type: application/json" \
    "https://api.scaleway.ai/v1/models"
```

When using the OpenAI Python SDK, the API key is set once during client initialization, and the SDK automatically manages the inclusion of the Authorization header in all API requests. 

> In contrast, when directly integrating with the Scaleway Generatie APIs, you are responsible for manually setting the Authorization header with the API key for each request to ensure proper authentication.

Reviewer: **Generative**

Reviewer 2: Good catch

---
meta:
  title: Using Generative APIs
  description: This page explains how to use Generative APIs
content:
  h1: Using Generative APIs
  paragraph: This page explains how to use Generative APIs
tags: generative-apis ai-data embeddings-api
dates:
  validation: 2024-08-28
  posted: 2024-08-28
---

## Access

> - Access to this service is restricted while in beta. You can request access to the product via [a form here](https://www.scaleway.com/en/betas/#generative-api).

Reviewer 3: ```suggestion
- Access to this service is restricted while in beta. You can request access to the product by filling out a form on the Scaleway's [betas page](https://www.scaleway.com/en/betas/#generative-api).
```

> - Access to this service is restricted while in beta. You can request access to the product via [a form here](https://www.scaleway.com/en/betas/#generative-api).

Reviewer 4: ```suggestion
- Access to this service is restricted while in beta. You can request access to the product via [the dedicated form](https://www.scaleway.com/en/betas/#generative-api).
```

> - Access to this service is restricted while in beta. You can request access to the product via [a form here](https://www.scaleway.com/en/betas/#generative-api).

Reviewer 5: ```suggestion
- Access to this service is restricted while in beta. [Fill out the participation form](https://www.scaleway.com/en/betas/#generative-api) to request access to the product.
```

> - A valid [API key](/identity-and-access-management/iam/how-to/create-api-keys/) is needed.

Reviewer 4: ```suggestion
- You need a valid [API key](/identity-and-access-management/iam/how-to/create-api-keys/).
```

## Authentication

All requests to the Scaleway Generative APIs must include an `Authorization` HTTP header with your API key prefixed by `Bearer`.

We recommend exporting your secret key as an environment variable, which you can then pass directly in your curl request as follows. Remember to replace the example value with your own API secret key.

```
export SCW_SECRET_KEY=720438f9-fcb9-4ebb-80a7-808ebf15314b
```

> Curl request:

Reviewer 4: ```suggestion
Run the following curl request once you have exported your environment variable:
```

In contrast, when directly integrating with the Scaleway Generative APIs, you are responsible for manually setting the Authorization header with the API key for each request to ensure proper authentication.

## Content types

Scaleway Generative APIs accept JSON in request bodies and returns JSON in response bodies. 
You will want to send the `Content-Type: application/json` HTTP header in your requests.

## Permissions

Permissions define the actions a user or an application can perform on Scaleway Generative APIs. They are managed using Scaleway’s [Identity and Access Management](/identity-and-access-management/iam/quickstart/) interface.

[Owner](/identity-and-access-management/iam/concepts/#owner) status or certain [IAM permissions](/identity-and-access-management/iam/concepts/#permission) allow you to perform actions in the intended Organization.

Querying AI models hosted by Scaleway Generative APIs will require any of the following [permission sets](/identity-and-access-management/iam/concepts/#permission-set):

- **GenerativeApisModelAccess**
- **GenerativeApisFullAccess**
- **AllProductsFullAccess**

## Projects

> 

Reviewer 3: seems like there's some content missing here

Reviewer 2: Lol yes, silly thing I forgot to save my local edit, so my commits got pushed without it. Adding with another push later today

Reviewer 2: done

> 

Reviewer 4: This part is missing, no?
